pyccel/ast/scipyext.py

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+""" Module containing objects from the scipy module understood by pyccel
+"""
+from numpy import pi
+from .core import Module, Import
+from .variable import Constant
+
+scipy_pi_const = Constant('float', 'pi', value=pi)
+
+scipy_mod = Module('scipy',
+        variables = (scipy_pi_const,),
+        funcs = (),
+        imports = [
+            Import('constants',Module('constants', (scipy_pi_const,), ()))
+            ])
```

Comment:
Does the `imports` argument in a `Module` constructor always refer to submodules?

Reply:
Yes

Reply:
No, they refer to anything imported in the module. E.g:
```
from numpy import empty

def f():
    a = empty(3)
```
Here the import is in the module not the function

Reply:
That's what I thought. Then it is not correct to say that the `scipy` module imports the `constants` module. In fact if you try
```python
import scipy
c = scipy.constants
```
you will get
```python
AttributeError: module 'scipy' has no attribute 'constants'
```

It seems to me that we are missing the concept of package, which may contain sub-packages and modules.

Reply:
Hmm, I think you are correct. But I am not sure we need to add packages now. In most packages you can access the sub-packages directly. E.g:
```
import numpy as np
r = np.random
```

Reply:
In your example the `numpy.__init__` module actually imports the sub-package `numpy.random`.

I also don't know if we need packages now. I am just worried that we are mixing two different concepts, namely "imported module" vs. "sub-package", which are a bit tricky to grasp in Python. This might become a problem in the future.

Reply:
Should we move this topic to a new discussion?

Reply:
Done. See https://github.com/pyccel/pyccel/discussions/1049